### PR TITLE
ESR 78 migration guide

### PR DESCRIPTION
**Pull request by ptomato**

### CODE REVIEW

File: examples/repl.cpp [hunk 1]
```diff
@@ -72,108 +74,6 @@ static void die(const char* why) {
   exit(1);
 }
 
-// The PrintError functions are modified versions of private SpiderMonkey API:
-// js/src/vm/JSContext.cpp, js::PrintError()
-
-enum class PrintErrorKind { Error, Warning, StrictWarning, Note };
-
-static void PrintErrorLine(const std::string& prefix, JSErrorReport* report) {
```
Comment (moztcampbell) [inline examples/repl.cpp:80]: :)

**Review by moztcampbell — APPROVED** [review]
This is great. It is all building and working locally for me too.

Nit: The versions numbers in the 'build' commit are wrong.

File: tools/jsopcode.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
-#!/usr/bin/env python -B
+#!/usr/bin/env python3 -B
```
Comment (hotsphink) [inline tools/jsopcode.py:1]: Doesn't this require `-S` too?

Comment (hotsphink) [reply tools/jsopcode.py:1]: Oh! This is imported directly from our tree. Ok.

**Review by hotsphink — APPROVED** [review]
Looks great to me.

File: tools/jsopcode.py (hunk 2 above)
Comment (ptomato) [reply tools/jsopcode.py:1]: It has some modifications to output GFM markdown, but otherwise yes.

**Comment by ptomato** [comment]
Thanks for the reviews especially during a difficult time! Sorry to hear the bad news this week. :heart:

**Pull request merged**